frontend/src/main/web/src/components/TextInput.jsx

```diff
@@ -0,0 +1,185 @@
+    focus: {
+      bdc: 'Bdc(pri):f'
+    }
+  }
+}
+
+class TextInput extends Component {
+  constructor () {
+    super()
+  }
+
+  _onBlur (e) {
+    const { onBlur } = this.props
+    if (onBlur) {
+      onBlur(e)
+    }
+  }
+  _onChange (e) {
+    const { onChange, onChangeText } = this.props
+    if (onChangeText) onChangeText(e.target.value)
+    if (onChange) onChange(e)
+  }
+
+  _onFocus (e) {
+    const { clearTextOnFocus, onFocus, selectTextOnFocus } = this.props
+    const node = ReactDOM.findDOMNode(this)
+    if (clearTextOnFocus) node.value = ''
+    if (selectTextOnFocus) node.select()
+    if (onFocus) onFocus(e)
+  }
+
+  _onSelectionChange (e) {
+    const { onSelectionChange } = this.props
+    const { selectionDirection, selectionEnd, selectionStart } = e.target
+    const event = {
+      selectionDirection,
+      selectionEnd,
+      selectionStart,
+      nativeEvent: e.nativeEvent
+    }
+    if (onSelectionChange) onSelectionChange(event)
+  }
+
+  _onKeyDown (e) {
+    const { onKeyDown } = this.props
+    if (onKeyDown) onKeyDown(e)
+  }
+
+  _onClear () {
+    const node = ReactDOM.findDOMNode(this)
+    node.value = ''
+  }
+
+  render () {
+    const {
+      id,
+      accessibilityLabel,
+      autoComplete,
+      autoFocus,
+      defaultValue,
+      editable = true,
+      keyboardType = 'default',
+      maxLength,
+      maxNumberOfLines,
+      multiline = false,
+      numberOfLines = 2,
+      onBlur,
+      onChange,
+      onChangeText,
+      onKeyDown,
+      onSelectionChange,
+      placeholder,
+      secureTextEntry = false,
+      theme,
+      value
+    } = this.props
+
+    let type
+
+    switch (keyboardType) {
+      case 'email-address':
+        type = 'email'
+        break
+      case 'numeric':
+        type = 'number'
+        break
+      case 'phone-pad':
+        type = 'tel'
+        break
+      case 'url':
+        type = 'url'
+        break
+    }
+
+    if (secureTextEntry) {
+      type = 'password'
+    }
+
+    const propsCommon = {
+      id: id,
+      'aria-label': accessibilityLabel,
+      autoComplete: autoComplete && 'on',
+      autoFocus,
+      className: flattenThemeClasses(classes, theme),
+      defaultValue,
+      maxLength,
+      onBlur: onBlur && this._onBlur.bind(this),
+      onChange: (onChange || onChangeText) && this._onChange.bind(this),
+      onFocus: this._onFocus.bind(this),
+      onSelect: onSelectionChange && this._onSelectionChange.bind(this),
+      onKeyDown: (onKeyDown) && this._onKeyDown.bind(this),
+      placeholder,
+      readOnly: !editable,
+      value
+    }
+
+    const propsMultiline = {
+      ...propsCommon,
+      maxRows: maxNumberOfLines || numberOfLines,
+      minRows: numberOfLines
+    }
+
+    const propsSingleline = {
+      ...propsCommon,
+      type
+    }
+
+    if (multiline) {
+      return <TextareaAutosize  {...propsMultiline} />
+    } else {
+      return <input {...propsSingleline}/>
```

Comment:
Move `propsMultiline` and `propsSingleline` into this conditional so it will only create the one it needs.